app/src/main/java/edu/grinnell/kdic/MainActivity.java

```diff
@@ -302,58 +295,64 @@ public void onClick(View v) {
-                                    .commit();
-                            break;
-                        default:
-                            break;
-                    }
-                }
-                // close the drawer after something is clicked
-                mDrawerLayout.closeDrawer(GravityCompat.START);
-                return true;
+                return setOnClickNavigationItem(menuItem);
             }
         });
     }
 
     /**
+     * Set response after clicking on items in the navigation drawer
+     */
+    public boolean setOnClickNavigationItem(MenuItem menuItem) {
+        if (mBackStack.peek() != menuItem.getItemId()) {
+            if (mBackStack.peek() == R.id.visualizer) {
+                hideVisualizeFragment();
+                mBackStack.pop();
+            }
+            mBackStack.push(menuItem.getItemId());
+
+            //Create fragment transaction to handle response to call from the navigation drawer
+            FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
+            switch (menuItem.getItemId()) {
+                case R.id.schedule:
+                    //generate schedule fragment after clicking on Schedule
+                    ft.replace(R.id.fragment, mScheduleFragment, ScheduleFragment.TAG)
+                            .addToBackStack(null)
+                            .commit();
+                    break;
+                case R.id.visualizer:
+                    //Generate now playing response
+                    showVisualizeFragment();
+                    break;
+                case R.id.favorites:
+                    //Generate favorite response
+                    ft.replace(R.id.fragment,
+                            mFavoritesFragment == null ? new FavoritesFragment() : mFavoritesFragment,
+                            FavoritesFragment.TAG)
+                            .addToBackStack(null)
+                            .commit();
+                    break;
+                case R.id.blog:
+                    //Generate blog fragment in response to the blog button
+                    ft.replace(R.id.fragment, new BlogWebViewFragment(), BlogWebViewFragment.TAG)
+                            .addToBackStack(null)
+                            .commit();
+                    break;
+                case R.id.about:
+                    //Generate about fragment in response to the about button
+                    ft.replace(R.id.fragment, new AboutFragment(), AboutFragment.TAG)
+                            .addToBackStack(null)
+                            .commit();
+                    break;
+                default:
+                    break;
+            }
+        }
+        // close the drawer after something is clicked
+        mDrawerLayout.closeDrawer(GravityCompat.START);
+        return true;
+    }
+    /**
      * Highlight selected item
      */
     public void updateNavigationView() {
```

Comment:
Should this method be public? If you're using it in just this class, you **should** make it `private`